cfg_mgmt/azure.py

```diff
@@ -0,0 +1,183 @@
+import copy
+import dataclasses
+import datetime
+import typing
+
+import msal
+import requests
+
+import cfg_mgmt
+import model
+import model.azure
+
+from cfg_mgmt.model import CfgQueueEntry
+
+AZURE_APP_ROOT_URL = 'https://graph.microsoft.com/v1.0/applications'
+
+
+@dataclasses.dataclass
+class PasswordCredential:
+    displayName: str
+    keyId: str
+    endDateTime: str
+    startDateTime: typing.Optional[str]
+    customKeyIdentifier: typing.Optional[str]
+    hint: typing.Optional[str]
+    secretText: typing.Optional[str]  # the actual password, if present
+
+    def __post_init__(self):
+        self.endDateTime = _to_datetime(self.endDateTime)
```

Comment:
while this works, I would prefer to mv `_to_datetime`'s code to before those usages
also: how about adding a "getter-method" doing the conversion (rather than altering the parsed attrs)?